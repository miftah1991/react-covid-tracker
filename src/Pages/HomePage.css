title {
  color: white;
  padding: 5px 20px;
  font-size: 1.5rem;
  text-align: left;
  background-color: var(--dark-pink);
}

.continents {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 150px 150px 150px;
}

.continents div {
  display: inline-grid;
  grid-template-columns: 50% 50%;
  position: relative;
}

.continents div:nth-child(2),
.continents div:nth-child(3),
.continents div:nth-child(6) {
  background-color: var(--dark-pink);
}

.mapDiv {
  text-align: center;
  height: 25vh;
  max-height: 25vh;
  margin: 10px 0 20px 0;
}

.map {
  max-width: 100%;
  width: 100%;
  height: 100%;
}

.navbar {
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 20px 0;
}

.navbar__img {
  width: 30px;
  height: auto;
}

.navbar__title {
  color: white;
  font-weight: 200;
}
