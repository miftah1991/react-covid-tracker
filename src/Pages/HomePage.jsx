import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllTimeData } from '../Redux/dashboard/dashboardReducer';
import { fetchContinents } from '../Redux/continents/continentsReducer';
import AllTimeStats from '../components/AllTimeStats/AllTimeStats';
import ContinentsDisplay from '../components/ContinentsDisplay/ContinentsDisplay';
import style from './HomePage.css';
import search from '../images/searchicon.svg';
import home from '../images/home.svg';

function HomePage() {
  const dispatch = useDispatch();
  const continents = useSelector((state) => state.continents);
  useEffect(() => {
    dispatch(fetchAllTimeData())
      .then(dispatch(fetchContinents()));
  }, []);
  return (
    <>
      <div className={style.navbar}>
        <img className={style.navbar__img} src={home} alt="Home" />
        <h1 className={style.navbar__title}>covid-19 stats</h1>
        <img className={style.navbar__img} src={search} alt="search" />
      </div>
      <div className={style.mapDiv}>
        <iframe className={style.map} title="covid map" src="https://coronavirus.app/map?embed=true" frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen />
      </div>

      <h2 className={style.title}>ALL TIME STATS</h2>
      <div>
        <AllTimeStats />
      </div>

      <h2 className={style.title}>STATS BY CONTINENT</h2>
      <div className={style.continents}>
        {continents.map((continent) => (
          <ContinentsDisplay key={continent.id} continent={continent} />
        ))}
      </div>
    </>
  );
}

export default HomePage;
