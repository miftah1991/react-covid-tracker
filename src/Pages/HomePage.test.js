import renderer from 'react-test-renderer';
import {
  render, screen,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/storeSetUp';
import HomePage from './HomePage';

it('snapshot of homepage', () => {
  const tree = renderer.create(
    <Router><Provider store={store}><HomePage /></Provider></Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('rendering page', () => {
  render(<Router><Provider store={store}><HomePage /></Provider></Router>);
  expect(screen.getByText('ALL TIME STATS')).toBeInTheDocument();
});