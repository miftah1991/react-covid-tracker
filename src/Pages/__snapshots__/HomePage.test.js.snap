// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of homepage 1`] = `
Array [
  <div
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Home"
        className="navbar__img"
        src="home.svg"
      />
    </a>
    <h1
      className="navbar__title"
    >
      covid-19 stats
    </h1>
    <img
      alt="search"
      className="navbar__img"
      src="searchicon.svg"
    />
  </div>,
  <div
    className="mapDiv"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      className="map"
      frameBorder="0"
      src="https://coronavirus.app/map?embed=true"
      title="covid map"
    />
  </div>,
  <h2
    className="title"
  >
    ALL TIME STATS
  </h2>,
  <div>
    <div
      className="dashboard"
    >
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        active cases
      </div>
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        today cases
      </div>
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        today deaths
      </div>
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        total cases
      </div>
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        total deaths
      </div>
      <div
        className="stat"
      >
        <span
          className="stat__number"
        >
          NaN
        </span>
        <br />
        total tests
      </div>
    </div>
  </div>,
  <h2
    className="title"
  >
    STATS BY CONTINENT
  </h2>,
  <div
    className="continents"
  />,
]
`;
